core/utils.go

```diff
@@ -128,23 +129,37 @@ func HighestAutoLevel(config *Config, allChanges []Change) (string, error) {
 			}
 
 			if kc.AutoLevel == "" {
+				// if we have a kind with no auto level and AllowSkippedEntries = false, then we return an error
 				return "", ErrMissingAutoLevel
 			}
 
+			if kc.AutoLevel == NoneLevel {
```

Comment:
I think we can set up a switch statement now, all the if statements are getting a bit verbose now.